src/lib/auth/strategies/wpcom/oauth.js

```diff
@@ -0,0 +1,200 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import wpComRequest from 'wpcom-xhr-request';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import { WPcomStrategy } from './index';
+import config from 'src/config';
+
+/**
+ * Module variables
+ */
+const debug = debugFactory( 'happychat-client:auth:wpcom:oauth' );
+const oAuth = require( 'wpcom-oauth-cors' )( config( 'oauth_client_id' ) );
```

Comment:
I'm not convinced we should do this within the library. My fear is that if we do it, this is what will happen on using the library in a host:

* an user visit myfantastichostthatuseshappy.chat, a host that embeds the happychat library :tada: 
* when the page loads, the library is initialized
* on initialization, because of this, it will redirect the user to the WordPress.com OAuth flow
* the user is on the WordPress.com page when she wanted to load myfantastichostthatuseshappy.chat instead :scream: 

I think this is not ideal and the reason I kept the auth outside the library.

But! In the future, we may want to do this differently and prepare the library to own authentication. When/If we do this it needs to be triggered by an user action (see https://github.com/Automattic/happychat-client/issues/125), for example: 

* you visit myfantastichostthatuseshappy.chat
* when the page loads, the library is initialized
* on initialization, the library knows the user is not authenticated, so instead of the contact/chat form it renders a button that says "authenticate with WordPress.com" or something like that
* on clicking that button, a new window is open that redirects the user to the OAuth flow
* when the OAUth flow is done, that new window is closed and the main page updated with the contact/chat form

This is essentially how other apps work.

So: I guess my concern is to do the OAUth dance on initialization. Unless we are ready to do that work, my recommendation would be to limit the authentication strategies to `wpcom-oauth-by-token` and `wpcom-proxy-iframe`. What do you think?

Reply:
Ah, by the way: I know the standalone example uses `wpcom-oauth`, but we can modify the example to pass the token to the library.

Reply:
I'll modify the example and remove this strategy.